packages/serverless/src/domain/use-case/document.ts

```diff
@@ -194,34 +198,30 @@ async function createDocument(
   const witness = await proveCreateDocument(
     databaseName,
     collectionName,
-    insertResult.docId,
-    fields,
+    docId,
+    Object.values(fields),
     compoundSession
   );
 
   return witness;
 }
 
 async function updateDocument(
-  databaseName: string,
-  collectionName: string,
-  actor: string,
+  permissionParams: TPermissionSudo<TParamCollection>,
   filter: FilterCriteria,
```

Comment:
FilterCriteria actually `Record<string, unknown>` and it doesn't help much, I think you can use Partial<Document> to perform better typing experience

Reply:
This is because `parseQuery` accepts an arbitrary Record:
https://github.com/orochi-network/zkDatabase/blob/refactor/document_related_enpoint_take_3/packages/serverless/src/domain/utils/document.ts

Although I added a TODO to consider a better type annotation.